tests/misc_test.go

```diff
@@ -190,3 +199,154 @@ var _ = Describe("SCC annotation", func() {
+			Name:      dvName,
+			Namespace: vm.GetNamespace(),
+		}, &cdiv1beta1.DataVolume{})
+
+		logObject(client.ObjectKey{
+			Name:      dvName,
+			Namespace: vm.GetNamespace(),
+		}, &core.PersistentVolumeClaim{})
+	})
+
+	table.DescribeTable("should be able to start VM", func(imageUrl string) {
+		const diskName = "disk0"
+
+		var always = kubevirtv1.RunStrategyAlways
+		var terminateGracePeriod int64 = 0
+		var pullMethodNode = cdiv1beta1.RegistryPullNode
+		var vmName = "test-vm-" + rand.String(5)
+
+		vm = &kubevirtv1.VirtualMachine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      vmName,
+				Namespace: strategy.GetNamespace(),
+			},
+			Spec: kubevirtv1.VirtualMachineSpec{
+				RunStrategy: &always,
+				Template: &kubevirtv1.VirtualMachineInstanceTemplateSpec{
+					Spec: kubevirtv1.VirtualMachineInstanceSpec{
+						Domain: kubevirtv1.DomainSpec{
+							Resources: kubevirtv1.ResourceRequirements{
+								Requests: core.ResourceList{
+									core.ResourceMemory: resource.MustParse("2Gi"),
+								},
+							},
+							CPU: &kubevirtv1.CPU{Sockets: 1, Cores: 1, Threads: 1},
+							Devices: kubevirtv1.Devices{
+								Rng: &kubevirtv1.Rng{},
+								Interfaces: []kubevirtv1.Interface{{
+									Name: "default",
+									InterfaceBindingMethod: kubevirtv1.InterfaceBindingMethod{
+										Masquerade: &kubevirtv1.InterfaceMasquerade{},
+									},
+								}},
+								Disks: []kubevirtv1.Disk{{
+									Name: diskName,
+									DiskDevice: kubevirtv1.DiskDevice{
+										Disk: &kubevirtv1.DiskTarget{Bus: "virtio"},
+									},
+								}},
+								NetworkInterfaceMultiQueue: pointer.BoolPtr(true),
+							},
+						},
+						TerminationGracePeriodSeconds: &terminateGracePeriod,
+						Networks:                      []kubevirtv1.Network{*kubevirtv1.DefaultPodNetwork()},
+						Volumes: []kubevirtv1.Volume{{
+							Name: diskName,
+							VolumeSource: kubevirtv1.VolumeSource{
+								DataVolume: &kubevirtv1.DataVolumeSource{
+									Name: vmName,
+								},
+							},
+						}},
+					},
+				},
+				DataVolumeTemplates: []kubevirtv1.DataVolumeTemplateSpec{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: vmName,
+					},
+					Spec: cdiv1beta1.DataVolumeSpec{
+						Source: &cdiv1beta1.DataVolumeSource{
+							Registry: &cdiv1beta1.DataVolumeSourceRegistry{
+								URL:        &imageUrl,
+								PullMethod: &pullMethodNode,
+							},
+						},
+						Storage: &cdiv1beta1.StorageSpec{
+							Resources: core.ResourceRequirements{
+								Requests: core.ResourceList{
+									core.ResourceStorage: resource.MustParse("10Gi"),
+								},
+							},
+						},
+					},
+				}},
+			},
+		}
+
+		Expect(apiClient.Create(ctx, vm)).To(Succeed())
+
+		// Wait for VM to start
+		Eventually(func() (bool, error) {
+			foundVm := &kubevirtv1.VirtualMachine{}
+			err := apiClient.Get(ctx, client.ObjectKeyFromObject(vm), foundVm)
+			if err != nil {
+				return false, err
+			}
+
+			return foundVm.Status.Ready, nil
```

Comment:
@akrejcir What does the Status.Ready means here? Does it mean the VM is booting, or that the guest is booted?

Reply:
The `status.Ready` is `true` when the pod has ready condition `true`. In this case, there is no readiness probe, so the pod is ready as soon as it's running. It does not mean anything about the VM.

I can add a readiness probe, that will check if the guest os has successfully booted.

Reply:
@akrejcir thanks for adding the ReadinessProbe!